suite-native/module-connect-device/src/components/PinFormProgress.tsx

```diff
@@ -5,14 +5,19 @@ import { useTranslate } from '@suite-native/intl';
 
 const MAX_DIGITS_DISPLAYED_AS_DOTS = 6;
 
-const dotStyle = prepareNativeStyle(utils => ({
-    width: utils.spacings.small,
-    height: utils.spacings.small,
-    borderRadius: utils.borders.radii.round,
-    borderColor: utils.colors.textDefault,
-    borderWidth: utils.borders.widths.small,
-    backgroundColor: utils.colors.textDefault,
-}));
+const dotStyle = prepareNativeStyle<{ isPinFormSubmitting: boolean }>(
+    (utils, { isPinFormSubmitting }) => {
+        const color = isPinFormSubmitting ? utils.colors.textDisabled : utils.colors.textDefault;
+        return {
+            width: utils.spacings.small,
+            height: utils.spacings.small,
+            borderRadius: utils.borders.radii.round,
+            borderColor: color,
+            borderWidth: utils.borders.widths.small,
+            backgroundColor: color,
+        };
+    },
+);
```

Comment:
you can use extend with condition for prepareNativeStyle:

```
const dotStyle = prepareNativeStyle<{ isPinFormSubmitting: boolean }>(
    (utils, { isPinFormSubmitting }) => ({
        width: utils.spacings.small,
        height: utils.spacings.small,
        borderRadius: utils.borders.radii.round,
        borderColor: utils.colors.textDefault,
        borderWidth: utils.borders.widths.small,
        backgroundColor: utils.colors.textDefault,
        extend: [
            {
                condition: isPinFormSubmitting,
                style: {
                    borderColor: utils.colors.textDisabled,
                    backgroundColor: utils.colors.textDisabled,
                },
            },
        ],
    }),
);
```

Reply:
cool, I didn't think of that, but in this case I like that color constant explicitly sets the same value for borderColor and backgroundColor.

Reply:
I don’t have a strong opinion on this. So up to you :). I just wanted to let you k iw in case you didnt.